### ISSUE TITLE
ZIO 2.0: Compositional `Platform`

### ISSUE THREAD

**Issue by jdegoes** [zio/zio#5476]
It is not possible to run part of an effect on a different platform, but this is a natural generalization of existing machinery.

```scala
ZIO.onPlatform(newPlatform) { effect }

zio.onPlatform(newPlatform)
```

This can be accomplished by making `platform` a `var` in `FiberContext`, adding a new operation to set the platform, and finally using `ensuring` in the implementation of `onPlatform` to ensure the old platform is restored.

**Comment by adamgfraser** [comment]
I will do this.

**Comment by adamgfraser** [comment]
Closed by #5477.
